### PR TITLE
feat: add audit api and audit authorization FPF-13263

### PR DESCRIPTION
**Pull request by frbattid**
# PR Summary

Added random data generation for Audit_Api and Audit_Authorization datasets.

## Detailed Description

Audit_Api and Audit_Authorization are audit-related datasets required in many developments. Synthetic samples are required in order to test those developments. Here we add such synthetic samples generation by means of the 4P Avro specification (`samples/datasets` folder) and the schema extensions (`samples/extensions` folder).

## Related Issues

https://jira.tid.es/browse/FPF-13263

## How Has This Been Tested

It has been manually tested by running the Avro Random Generator CLI, producing some synthetic samples:

```
lanuza/pipelines/run.sh lanuza/scripts/build-data.sh --dataset-id=Audit_Api --dataset-version=2.7.0 --records=20 --datasets-in=samples/datasets --datasets-extensions-in=samples/extensions --out=datasets
```

```
lanuza/pipelines/run.sh lanuza/scripts/build-data.sh --dataset-id=Audit_Authorization --dataset-version=1.6.0 --records=20 --datasets-in=samples/datasets --datasets-extensions-in=samples/extensions --out=datasets
```

### CODE REVIEW

File: samples/extensions/Audit_Api/v2.7.0/extensions.json [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+          "type": "record",
+          "name": "USER_INFO_RECORD",
+          "fields" : [
+            {
+              "name": "USER_ID",
+              "type": {
+                "type": "string",
+                "x-fp-user-id": true,
+                "arg.properties": {
+                  "options" : [ "user1", "user2" ]
```
Comment (frbattid) [inline samples/extensions/Audit_Api/v2.7.0/extensions.json:75]: Debería poner un string similar a un USER_ID "real", en lugar de "user1". Esta duda aplica a casi todos los campos :D

**Review by locl95 — APPROVED** [review]

File: samples/datasets/Audit_Api/v2.7.0/Audit_Api_2.7.0.avsc [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                  }
+                ], "default": null, "doc": "Scopes returned within the access_token when externally login."
+              }
+            ]
+          }
+        ], "default": null, "doc": "MAY_ACT claim."
+      },
+      { "name": "CLIENT_IP_ADDRESS", "type": ["null", { "type": "string", "logicalType": "ipv4", "x-fp-data-protection": "pseudonymize"}], "default": null, "doc": "The client IP address." }
+    ]
+}
```
Comment (yaizaff) [inline samples/datasets/Audit_Api/v2.7.0/Audit_Api_2.7.0.avsc:87]: añadir línea en blanco al final

File: samples/datasets/Audit_Authorization/v1.6.0/Audit_Authorization_1.6.0.avsc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                    "type": "array", "items": "string"
+                  }
+                ], "default": null, "doc": "Scopes returned within the access_token when externally login."
+              }
+            ]
+          }
+        ], "default": null, "doc": "MAY_ACT claim."
+      }
+    ]
+  }
```
Comment (yaizaff) [inline samples/datasets/Audit_Authorization/v1.6.0/Audit_Authorization_1.6.0.avsc:93]: añadir línea en blanco al final

File: samples/extensions/Audit_Api/v2.7.0/extensions.json (hunk 1 above)
Comment (yaizaff) [reply samples/extensions/Audit_Api/v2.7.0/extensions.json:75]: Yo creo que aquí da un poco igual. La persona que vaya generar el avro no sabemos que datos va a necesitar.

File: samples/datasets/Audit_Authorization/v1.6.0/Audit_Authorization_1.6.0.avsc (hunk 3 above)
Comment (frbattid) [reply samples/datasets/Audit_Authorization/v1.6.0/Audit_Authorization_1.6.0.avsc]: https://github.com/Telefonica/avro-random-generator/pull/36/commits/91985b5219dfd179b6079c62fea3145054941d60

File: samples/datasets/Audit_Api/v2.7.0/Audit_Api_2.7.0.avsc (hunk 2 above)
Comment (frbattid) [reply samples/datasets/Audit_Api/v2.7.0/Audit_Api_2.7.0.avsc]: https://github.com/Telefonica/avro-random-generator/pull/36/commits/91985b5219dfd179b6079c62fea3145054941d60

**Review by yaizaff — APPROVED** [review]

**Pull request merged**